### PR TITLE
Keithley 6500 parameters had invalid SCPI commands

### PR DESCRIPTION
**Pull request by qSaevar**
Changes proposed in this pull request:
In Keithley_6500 driver fix parameters:

- trigger_source
- trigger_delay
- display_enable

@astafan8

### CODE REVIEW

**Comment by jenshnielsen** [comment]
It looks like the dmm supports compatibility mode with old dmms. I suspect the original driver was written to support this

**Comment by qSaevar** [comment]
@jenshnielsen It is stated in the class docstring that this is based on the keithley 2000 (where the SCPI commands are valid) and that this is a bet-version. Don't know if there has been a 6500 in the past where the three parameters that I changed worked. My guess is not.

File: qcodes/instrument_drivers/tektronix/Keithley_6500.py [hunk 1]
```diff
@@ -109,10 +109,15 @@ def __init__(self, name, address, reset_device=False, **kwargs):
...
-                           get_cmd='DISP:ENAB?',
-                           set_cmd='DISP:ENAB {}', set_parser=int, vals=Bool())
+        self.add_parameter('display_backlight',
+                           docstring='Control the brightness of the display '
+                                     'backligt. OFF turns the display off and'
+                                     'BLACkout also turns off indicators and '
+                                     'key lights on the device.',
+                           get_cmd='DISP:LIGH:STAT?',
+                           set_cmd='DISP:LIGH:STAT {}',
+                           vals=Enum('ON100', 'ON75', 'ON50', 'ON25', 'OFF',
```
Comment (astafan8) [inline qcodes/instrument_drivers/tektronix/Keithley_6500.py]: i'd recommend using val_mapping here and give user-friendly names to each value.

File: qcodes/instrument_drivers/tektronix/Keithley_6500.py [hunk 2]
```diff
@@ -121,17 +126,32 @@ def __init__(self, name, address, reset_device=False, **kwargs):
...
+                               docstring='Set and read trigger delay for '
+                                         'timer %i.' % trigger,
+                               get_parser=float,
+                               get_cmd='TRIG:TIM%i:DEL?' % trigger,
+                               set_cmd='TRIG:TIM%i:DEL {}' % trigger,
+                               unit='s', vals=Numbers(min_value=0,
+                                                      max_value=999999.999))
+
+            self.add_parameter('trigger%i_source' % trigger,
+                               docstring='Set, and read, the event that sets'
```
Comment (astafan8) [inline qcodes/instrument_drivers/tektronix/Keithley_6500.py]: this docstring is confusing: "set and read the event"?

**Comment by jenshnielsen** [comment]
@qSaevar Yes my point is that the 6500 can be put into a 2000 compatibility mode. I think that could be what this driver is targeting. See  #1533 for something similar example for the the 2450

**Comment by qSaevar** [comment]
@jenshnielsen Ok, I see you point. So, solutions: Flag in the constructor for compatibility mode set by the user? Second driver for the full feature support? Some auto detection for the mode the device is in?

File: qcodes/instrument_drivers/tektronix/Keithley_6500.py (hunk 2 above)
Comment (qSaevar) [reply qcodes/instrument_drivers/tektronix/Keithley_6500.py]: It is confusing, fixed.

File: qcodes/instrument_drivers/tektronix/Keithley_6500.py (hunk 1 above)
Comment (qSaevar) [reply qcodes/instrument_drivers/tektronix/Keithley_6500.py]: Done

**Comment by jenshnielsen** [comment]
@qSaevar Yes I think we should test which mode it's in and error appropriately if it's in the wrong mode. I support using the default mode and not any compatibility mode

**Comment by qSaevar** [comment]
@jenshnielsen ok, so I'll add something like:
if language not SCPI:
  raise MeaningfulError
At the end of the constructor.

**Comment by qSaevar** [comment]
I did add a check for the correct command set.

**Comment by qSaevar** [comment]
@astafan8 I addressed your comments, @jenshnielsen concerns and updated the branch. Is there anything else that needs to be done?

File: qcodes/instrument_drivers/tektronix/Keithley_6500.py [hunk 3]
```diff
@@ -149,6 +177,12 @@ def __init__(self, name, address, reset_device=False, **kwargs):
         self.write('FORM:DATA ASCII')
         self.connect_message()
 
+        command_set = self.ask_raw('*LANG?')
```
Comment (astafan8) [inline qcodes/instrument_drivers/tektronix/Keithley_6500.py]: i think this code should go above the `self.write` call. perhaps even to the top of the __init__ (right after the super __init__ call).

Also, no need for using ask_raw - just use ask :)

Comment (qSaevar) [reply qcodes/instrument_drivers/tektronix/Keithley_6500.py]: ✔️

Comment (qSaevar) [reply qcodes/instrument_drivers/tektronix/Keithley_6500.py]: done

**Pull request merged**